doc/source/development/dispatching.rst

```diff
@@ -0,0 +1,93 @@
+Dispatching
+===========
+
+The dispatching mechanism allows users to use alternative implementations of the algorithms
+available in scikit-image. The alternative implementations are provided by separate
+Python packages maintained outside of scikit-image. Example use-cases are support for array
+libraries other than Numpy or optimised implementations for particular hardware.
+
+This section of the documentation describes how to create an alternative implementation (a "backend")
+and how the dispatching mechanism in scikit-image works.
+
+.. important::
+    Dispatching support and infrastructure is experimental and is not ready for production.
+    It's made available as an early draft so that developers can gain experience
+    with the system.
+
+    Expect the dispatching API to change without notice.
+
+
+Implementing a new backend
+--------------------------
+
+An alternative implementation ("backend") is a good place to provide optimized implementations
+for particular hardware, support array libraries other than Numpy or explore novel ideas that
+are not (yet) a good fit for the core scikit-image library.
+
+To create a backend you have to create a new Python package that registers two particular
+`entrypoints <https://packaging.python.org/en/latest/specifications/pyproject-toml/#entry-points>`_. The first entrypoint called `skimage_backends` should resolve to a namespace
+that contains two functions: `can_has(name, *args, **kwargs)` and `get_implementation(name)`.
+The second entrypoint called `skimage_backend_infos` should resolve to a function with no
+arguments that returns an instance of the `skimage.util._backends.BackendInformation` class
+or something that behaves like it.
+
+
+The `skimage_backend_infos` entrypoint
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The information your backend provides via this entrypoint is used to get things like its
+name, homepage and which functions it implements. The main requirement for this entrypoint
+is that it is fast to import and has no dependencies other than `scikit-image`.
```

Comment:
The language is tricky :-/ Even just what to call an "entry point" seems to be overloaded.

I think there are three steps:
1. discover an entry point (call `importlib.metadata.entry_points()` + `select() or `get()` (in older Python versions)
2. loading an entry point (call `some_ep.load()`). In the docs for `.load` it is referred to as "resolving the value"
3. doing something with the return value of step (2)

Steps 2 and 3 are the ones we need to be fast and can be influenced by the creator of a backend. For example they could put a expensive import statement in the file that contains their function that returns the information about the backend. Or they could try and discover the list of functions the backend supports dynamically when the function is called.

So I think "load" is a better word, especially for step (2). However I want to refer to both (2) and (3). And I wonder if people who are not entry point geeks know that "loading an entry point" means that is imported. That is how I arrived at "import" in the hopes that it conveys to the reader that "importing the function you provide has to be fast".

But yeah, not sure what is best to write.

Reply:
I think "...loading the backend with the entry-point..." would a better phrasing because that would be consistent with the [official docs](https://packaging.python.org/en/latest/specifications/entry-points/):

> Applications can use entry points to load plugins; e.g. Pygments (a syntax highlighting tool) can use additional lexers and .....

here, plugins == backends

Additionally, in the first point, we are not discovering the entry-point, we are discovering the backend packages using the entry-point specification. And in the second point we are loading the **object referred** by the entry-point(not the entry-point). LMK if that makes sense 2u.